### PR TITLE
Run PR actions on PRs from forks

### PR DESCRIPTION
**Pull request by Matyrobbrt**
*TL;DR: This implements a workaround for a GitHub limitation.*

In order to run PR opened and reviewed actions we need the autoforge App ID and key, which are *usually* given as secrets.  
However, when workflows run on forks (like the `pull_request`, `pull_request_review` and `pull_request_review_comment` events, which run on the top commit of the branch to merge), we do not have access to secrets, as per the [documentation](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow):
> With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository.

# Bypassing
In order to bypass this, we are going to use `workflow_run` and action artifacts to "simulate" events, needing 2 workflows:
- a workflow which will run on `pull_request_(review/review_comment)`: this workflow will upload its event payload json as an artifact;
- a workflow which will listen for completion of the above workflow (using `workflow_run`), and which will fake the event by replacing the payload with the one from the uploaded artifact.
Unfortunately, we cannot use `workflow_dispatch` as the `GITHUB_TOKEN` from forks is **read-only** so we cannot dispatch an event using the API.

This fix can be seen in action in https://github.com/MinecraftForge/Actionable/pull/12.

### CODE REVIEW

**Comment by FishyWis** [comment]
hello I'm FishyWis do u go to school if you do go to history spot and on the top of it when you go in it it says activities click that...

**Comment by FishyWis** [comment]
it has a bunch of games

**Comment by yorulmazzburcin-spec** [comment]
Lkhbljhbkjgv jygfıugf

**Comment by yorulmazzburcin-spec** [comment]
I…… am,steaveeee
